src/functions-and-arrays.js

```diff
@@ -1,19 +1,144 @@
 // Iteration #1: Find the maximum
+// function maxOfTwoNumbers(num1,num2){
+//   if (num1==num2){
+//     return num1;
+//   }else if (num1 > num2){
+//     return num1;
+//   }else {
+//     return num2;
+//   }
+// }
+function maxOfTwoNumbers(num1, num2) {
+  return num1 > num2 ? num1 : num1 == num2 ? num2 : num2;
+}
+
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpothhhhhhhhh'];
+function findLongestWord(arr) {
+  if (arr.length === 0) {
+    return null;
+  }
+  var comparison = arr[0];
+  for (var i = 0; i < arr.length; i++) {
+    if (comparison.length < arr[i].length) {
+      comparison = arr[i];
+    }
+  }
+  return comparison;
+}
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+// function sumNumbers(arr){
+//   if (arr.length == 0){
+//     return 0;
+// } 
+// let sum = 0;
+// for (let i=0; i < arr.length; i++) {
+//  sum += arr[i];
+// } 
+// return sum
+// }
+// //3.1
+// function sum(arr) {
+//   if (arr.length == 0){
+//     return 0;
+// } 
+// let sum = 0;
+// for (let i=0; i < arr.length; i++) {
+//  sum += arr[i];
+// } 
+// return sum
+// }
+function sumNumbers(numbers) {
+  if (numbers.length === 0) {
+    return 0;
+  }
+  var sum = () => numbers.reduce((acc, currentValue) => acc + currentValue);
+  return sum();
+}
+function sum(numbers){
```

Comment:
You could make it with a switch for example:

switch (typeof numbers[i]) {
      case "number":
        total+= numbers[i];
        break;
      case "string":
        total += numbers[i].length;
        break;
      case "boolean":
        if (numbers[i] === true) {
          total += 1;
        } else {
          continue;
        };
        break;
      default:
         throw new Error("Unsupported data type sir or ma'am");
    }
  }
  return total;

or with forEach:

  let sum = 0;
  numbers.forEach( function (el) {
    if (typeof el === 'string') {
      sum += el.length;
    }
    else if (typeof el === 'object') {
      throw new Error("Unsupported data type sir or ma'am")
    } 
    else {
      sum += el
    };
  })

  return sum;